### DISCUSSION TITLE
How to start open-block-explorer on WSL

### DISCUSSION THREAD

**Discussion by petervalencic** [telosnetwork/open-block-explorer#16] [Q&A] [answered]
Hi, 
can someone explain to me how to run the page?
I have cloned the code to my local PC (WSL) compiled it with yarn and installed quasar cli..
The node version I'm using is v.12.22.1.
After the build, I have moved to ./dist/spa and executed "quarar serve ."
When going to localhost:4000/  I can see the request for index.html but got empty HTML page 

![image](https://user-images.githubusercontent.com/10866388/152999465-32b90fab-c8b0-4152-86a3-3a3bd4db52bd.png)

Can someone tell me how to run the app?  (sorry I come from Java world ;) )

BR

**Comment by donnyquixotic** [comment]
Hi Peter, 
    It looks like you are trying to run the production build.  The url should not contain `index.html`  and you may not have run the correct command for production build before serving.  See steps below for both development & production.

development server:
- run `yarn` to build project
- run `yarn dev`  (or `quasar dev`)
- view app at `localhost:8080`

production build:
- run `yarn` to build project
- run `yarn build` in the project root
- navigate to the spa directory `cd dist/spa`
- run `quasar serve`
- view app at `localhost:4000` (no `index.html`).

**Reply by donnyquixotic** [reply]
*README updated with additional production steps*

**Reply by petervalencic** [reply]
Hi,
this is the attempt with yarn dev:
![image](https://user-images.githubusercontent.com/10866388/153131070-0fe12178-7cec-4aad-a6bd-aabd4d9e4f8a.png)

The page as you can see is just a blank page..
The view code shows:
![image](https://user-images.githubusercontent.com/10866388/153131176-c2e1ef04-01b8-43d0-9ae2-e2fce96bac0d.png)


If I run "quasar dev" I got some exceptions:
![image](https://user-images.githubusercontent.com/10866388/153131604-676822af-6cae-40bc-a84c-74b8ebcfa97f.png)


I'm using WSL on windows 10 ..

**Reply by donnyquixotic** [reply]
In the first two screen shots  I notice the `index.html` is in the url address bar, is it auto-populating that? I am able to re-create the problem by _adding_ that in my browser. Remove `/index.html` and see if it loads at just `localhost:8080` (and report any errors in console).  In the last screen shot you are attempting to run dev mode from the production build directory which will not work (see above for production steps). `quasar dev` (or `yarn dev`) should only be run in project root and only applies to dev mode.

**Reply by petervalencic** [reply]
Sorry... Here I have run the ```yarn dev``` from the project root. The browser is opened but the page is empty.. 

![image](https://user-images.githubusercontent.com/10866388/153208963-3b476944-868b-4cc9-b79b-d91ff191f40d.png)

![image](https://user-images.githubusercontent.com/10866388/153209024-c6d54621-a155-49f7-adc8-351cb474c4e4.png)

or 

![image](https://user-images.githubusercontent.com/10866388/153209334-b6d7559f-07ec-453b-bd4d-5e21c4caa4b1.png)

**Comment by donnyquixotic** [comment]
@petervalencic I was able to reproduce the bug. I _believe_ you are missing necessary environment variables file, '.env'. There should have been an `.env.example` file in the project root (.env is ignored by git) which was removed by mistake. I've re-added it to the develop branch. 

- pull develop into your branch `git pull origin develop`
- rename the .env.example file `cp .env.example .env`
 
This _should_ resolve your issue. Please let me know either way.

**Reply by petervalencic** [reply, marked as answer]
✌
![image](https://user-images.githubusercontent.com/10866388/153416782-f9c27ed0-0ff1-4273-9a60-18e9add4c2f9.png)
